### DISCUSSION TITLE
使用 map 调用全部 plugin hook 给 pluginImpls 赋值的方式是否违背了 hook 的调用原则

### DISCUSSION THREAD

**Discussion by me-for-i** [alibaba/hooks#2503] [Q&A]
如图所示
<img width="1440" alt="Snipaste_2024-03-15_10-57-58" src="https://github.com/alibaba/hooks/assets/40806323/dd885338-8669-45f4-a66e-061eabace5e3">

**Comment by liuyib** [comment]
@crazylxr 见哥有空看下这个呢

**Comment by crazylxr** [comment]
@me-for-i 你认为违反了 hook 的哪条原则？

**Reply by me-for-i** [reply]
hook不能作为一个普通函数在循环结构中调用吧，plugins 的数量和顺序是可以外部受控的，那么 hook 的调用顺序是否存在无法保障的可能
